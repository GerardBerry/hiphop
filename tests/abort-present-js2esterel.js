var js2esterel = require("../js2esterel.js");
var example = require("./abort-present.js");

console.log(example.prg.esterel_code());
