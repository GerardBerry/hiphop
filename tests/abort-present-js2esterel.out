module abortpresent:
input I;
output J, K, V;
loop
    abort
        emit J;
        pause;
        emit V;
        pause
    when I;
    present I then
        emit K
    end present
end loop
end module
